frontend/src/actions/toastActions.js

```diff
@@ -2,11 +2,11 @@ import * as TYPES from "./types";
 
 import { uid } from "@/utils/helper";
 
-export const showFailureToast = () => async (dispatch) => {
+export const showFailureToast = (message = null) => async (dispatch) => {
   const toast = {
     variant: "danger",
     title: "Something went wrong",
-    message: "Please try again later",
+    message: message ? message : "Please try again later",
```

Comment:
Sigh. I was tired of the static generic errors in the UI and tried to improve it without massively overhauling the lower layers. Given the various patterns FastAPI returns under the `detail` field, I wasn't entirely successful anyway, so rather than argue I guess I'll just drop this entirely. Varshini's massively reworking segments of the UI code, so worrying about merging in another separate stream of changes probably doesn't make sense.